DiverseSelector/metric.py

```diff
@@ -23,25 +23,292 @@
 
 """Metric calculation module."""
 
+from typing import Any
+
 import numpy as np
-from scipy.spatial.distance import cdist
+from scipy.spatial.distance import cdist, squareform
+from sklearn.metrics import pairwise_distances
+
+from DiverseSelector.utils import sklearn_supported_metrics
 
 __all__ = [
     "pairwise_dist",
     "compute_diversity",
+    "distance_to_similarity",
+    "pairwise_similarity",
+    "pairwise_similarity_bit",
+    "tanimoto",
+    "cosine",
+    "dice",
+    "bit_tanimoto",
+    "bit_cosine",
+    "bit_dice",
 ]
 
 
+class ComputeDistanceMatrix:
+    """Compute distance matrix.
+
+    This class is just a demo and not finished yet."""
+
+    def __init__(self,
+                 feature: np.ndarray,
+                 metric: str = "euclidean",
+                 n_jobs: int = -1,
+                 force_all_finite: bool = True,
+                 **kwargs: Any,
+                 ):
+        """Compute pairwise distance given a feature matrix.
+
+        Parameters
+        ----------
+        feature : np.ndarray
+            Molecule feature matrix.
+        metric : str, optional
+            Distance metric.
+
+        """
+        self.feature = feature
+        self.metric = metric
+        self.n_jobs = n_jobs
+        self.force_all_finite = force_all_finite
+        self.kwargs = kwargs
+
+    def compute_distance(self):
+        """Compute the distance matrix."""
+        built_in_metrics = [
+            "tanimoto",
+            "modified_tanimoto",
+
```

Comment:
This would lead to an error when running `tox` locally. So a good practice is to run `tox` locally to make sure things work and then push GitHub. Each commit should be small and having one commit with many lines of codes is a disaster.